tests/forms_tests/tests/test_formsets.py

```diff
@@ -877,6 +884,67 @@ def test_more_initial_than_max_num(self):
 <td><input id="id_form-2-name" name="form-2-name" type="text" value="Jack and Coke"></td></tr>"""
         )
 
+    def test_default_absolute_max(self):
+        """
+        The default absolute max is 2*formsets.DEFAULT_MAX_NUM
+        """
+        data = {
+            'form-TOTAL_FORMS': '2505',
+            'form-INITIAL_FORMS': '0',
+            'form-MAX_NUM_FORMS': '0',
+        }
+
+        # The default absolute_max is max_num + DEFAULT_MAX_NUM (which is
+        # 2*DEFAULT_MAX_NUM if max_num isn't specified)
+        formset = FavoriteDrinksFormSet(data=data)
+        self.assertFalse(formset.is_valid())
+        self.validate_formset_number_of_forms(formset, 2 * formsets.DEFAULT_MAX_NUM)
+
+    def test_explicit_absolute_max_higher_than_default(self):
+        """
+        absolute_max can now be explicitly specified via kwarg to formset_factory.
+        This behavior was changed from absolute_max always being max_num + 1000 in the patch for #20347
+        """
+        data = {
+            'form-TOTAL_FORMS': '2505',
+            'form-INITIAL_FORMS': '0',
+            'form-MAX_NUM_FORMS': '0',
+        }
+
+        # absolute_max can now be explicitly set, not being affected by
+        # max_num or DEFAULT_MAX_NUM at all
+        HigherLimitFavoriteDrinksFormSet = formset_factory(FavoriteDrinkForm, max_num=1, absolute_max=3000)
+        formset = HigherLimitFavoriteDrinksFormSet(data=data)
+        self.assertTrue(formset.is_valid())
+        self.validate_formset_number_of_forms(formset, 2505)
+
+        # verify that absolute_max still does provide a hard limit
```

Comment:
Please omit prefixes like "Verify that" since all tests verify things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).